src/main/js/controls/Graph/helpers/helpers.js

```diff
@@ -419,20 +453,70 @@ const createVGridHandler = (gridSVG, axis, style, config) => {
         .classed(style, true)
         .call(translateVerticalGrid(axis, config));
 };
+/**
+ * calculates the character limit of label with respect to axis length.
+ *
+ * @private
+ * @param {string} config - config object derived from input JSON
+ * @returns {object} character limit for both and y-axis
+ */
+const getAxesLabelCharacterLimits = (config) => ({
+    /*
+     *  We can fit 1 Uppercase character for every ≈7px. To fit ellipses with truncated text, we can assign 8px per character for y and y2-axis.
+     *  And if we use 8px per character, there will be some space left on x-axis. Inorder to use maximum space on x-axis,
+     *  we can assign less than 8px(22/3 i.e ≈7.3) per character in x-axis.
+     *
+     */
+    xAxisLimit:  getXAxisWidth(config) / 7.33,
+    yAndY2AxisLimit: config.height / 8
+});
+/**
+ * Adds onClick event for each label in Axes.
+ * Criteria:
+ *  * Text needs to have a valid function for onClick
+ *  * Label should exceed the calculated character limit with respect to axis length
+ *
+ * @private
+ * @param {d3.selection} canvasSVG - d3 selection node of canvas svg
+ * @param {string} className - class name of axis
+ * @param {object} axisObj - Axis object provided by the consumer
+ * @param {number} charLimit - character limit of label in axis
+ * @param {string} axisType - type of axis
+ * @returns {object} d3 svg path
+ */
+const addLabelEventHandler = (
+    canvasSVG,
+    className,
+    axisObj,
+    charLimit,
+    axisType
+) =>
+    canvasSVG.selectAll(`.${className} text`).each(function (displayVal) {
+        shouldTruncateLabel(axisObj.label, charLimit)
+            ? d3
+                  .select(this)
+                  .style("cursor", "pointer")
+                  .on("click", () => {
+                      utils.isDefined(axisObj.onLabelClick)
```

Comment:
Probably, we can check this condition as well `utils.isFunction(labelObj.onLabelClick)`

```js
utils.isDefined(labelObj.onLabelClick) && utils.isFunction(labelObj.onLabelClick)
```

Reply:
What I have observed here, here axisObj has `onLabelClick` and Gantt chart has `onClick`. 

Both should have the same name?

Reply:
[fixed](https://github.com/cerner/carbon-graphs/pull/242/commits/3d75d6b1109fb3da7bad04f51b1228bd6bcaaf93#diff-608946a83d2c753fa2af8bbb05dc72d2R500) Added function check for the labelObj.onLabelClick.

The reason why we have different name is, gantt chart has separate object for label. In that case we can just have onClick which can clearly states that  onClick is for label but in graphs we do not have separate object for label that is the reason why we named onLabelClick.